azure-toolkit-libs/azure-toolkit-appservice-lib/src/main/java/com/microsoft/azure/toolkit/lib/appservice/model/PricingTier.java

```diff
@@ -90,8 +87,10 @@ public boolean equals(Object o) {
         if (!(o instanceof PricingTier)) {
             return false;
         }
-        final PricingTier that = (PricingTier) o;
-        return Objects.equals(tier, that.tier) && Objects.equals(size, that.size);
+        // todo: update the implement once we find solution to serialize parameters without public setter/constructor
+        final PricingTier current = PricingTier.fromString(this.tier, this.size);
+        final PricingTier target = PricingTier.fromString(((PricingTier) o).tier, ((PricingTier) o).size);
```

Comment:
why not use ((PricingTier) o) directly

Reply:
Reparse here in case user modify value manually